src/protocols/radius/encode.c

```diff
@@ -710,12 +709,46 @@ static ssize_t encode_value(uint8_t *out, size_t outlen,
 	return out_p - out;
 }
 
-static ssize_t attr_shift(uint8_t const *start, uint8_t const *end,
+/** Encodes the data portion of an attribute into a dbuff
+ *
+ * Fully migrating away from the original encode_value() will require a revision of
+ * fr_struct_to_network() and revising the fr_encode_value_t type.
+ */
+static ssize_t encode_value_dbuff(fr_dbuff_t *dbuff,
```

Comment:
I don't think `encode_value` is used outside of this compilation unit, so there's no reason to wrap it?

Reply:
Sorry; I thought replying to the email would show up here. The ptr/len version of encode_value() has to stay around because it's passed as a parameter to fr_struct_to_network(), which others use (and which, because it takes function pointer parameters, isn't easy to wrap); I wrote the one layer around it rather than doing it in each caller in src/protocols/radius/encode.c.